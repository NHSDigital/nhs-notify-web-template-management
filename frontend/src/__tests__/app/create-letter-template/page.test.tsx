/**
 * @jest-environment node
 */
import CreateLetterTemplatePage from '@app/create-letter-template/page';

describe('CreateLetterTemplatePage', () => {
  beforeEach(jest.resetAllMocks);

  it('should render CreateLetterTemplatePage', async () => {
    const page = await CreateLetterTemplatePage();

    expect(page).toMatchSnapshot();
  });

  it('returns 404 when letters feature flag is not enabled', async () => {
    delete process.env.NEXT_PUBLIC_ENABLE_LETTERS;

    await expect(CreateLetterTemplatePage()).rejects.toThrow('NEXT_NOT_FOUND');
  });
});
