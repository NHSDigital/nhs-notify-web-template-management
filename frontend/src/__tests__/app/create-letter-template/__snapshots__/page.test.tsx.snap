// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateLetterTemplatePage should render CreateLetterTemplatePage 1`] = `
<LetterTemplateForm
  initialState={
    {
      "language": "ENGLISH",
      "letterType": "STANDARD",
      "name": "",
      "pdfTemplateInputFile": "",
      "templateStatus": "NOT_YET_SUBMITTED",
      "templateType": "LETTER",
    }
  }
/>
`;
